P202-Training.Domain/RoleService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using System.Collections.Generic;
+using P202.Training.Data;
+using AutoMapper;
+
+namespace P202.Training.Domain
+{
+    public class RoleService : IRoleService
+    {
+        private RoleRepository _repository;
+        public void CreateRole(Role role)
+        {
+            if (role == null)
+                return;
+            Mapper.Initialize(cfg => cfg.CreateMap<Role, Role>());
+            var mapRole = Mapper.Map<Role>(role);
+            var roleRepository = new RoleRepository(new NHibernateSessionManager());
```

Comment:
You shouldn't be instantiating the repository since it's a dependency. Instead you should have it injected in the constructor. Let the DI engine handle the instantiation